spec/services/publishers/ats_api/update_vacancy_service_spec.rb

```diff
@@ -221,7 +225,7 @@
   end
 
   context "when a vacancy with the same external reference exists" do
-    let!(:existing_vacancy) do
+    let(:existing_vacancy) do
```

Comment:
personally I prefer the existing pattern - IMHO it's less prone to confusion during code changes. I actually fixed a bug recently where the test had a let w/o a bang and no 'phantom reference' so the test passed even though the code was incorrect because the side-effect didn't take place. I have used the 'create and find' pattern in the past which is maybe clearer?

Reply:
I also favour the `let!` to be fair. I only did this because I needed to redefine the `existing_vacancy` in an inner context and `let!` wasn't allowing it.

Re-visiting this, I think this is just a sign that the `contexts` for deleted/non-deleted should be defined in parallel, not one inside another. 

Going to fix it 👷‍♂️